src/elm/type.js

```diff
@@ -29,14 +29,13 @@ class As extends Expression {
     if (arg == null) {
       return null;
     }
-    if (typeof arg._is !== 'function' && !isSystemType(this.asTypeSpecifier)) {
-      // We need an _is implementation in order to check non System types
-      // If this is not found then we should just return the arg to match old functionality.
-      return arg;
-    }
```

Comment:
Turns out we did not need this because it more or less shakes out in the implementation of `ctx.matchesTypeSpecifier`.